tests/framework/extensions/etcdsnapshot/config.go

```diff
@@ -0,0 +1,19 @@
+package etcdsnapshot
+
+import (
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+)
+
+const (
+	ConfigurationFileKey = "snapshotInput"
+)
+
+type ETCD struct {
+	ETCD            *rkev1.ETCD `json:"etcd" yaml:"etcd"`
+	SnapshotRestore string      `json:"snapshotRestore" yaml:"snapshotRestore"`
+}
+
+type Config struct {
+	UpgradeKubernetesVersion string `json:"upgradeKubernetesVersion" yaml:"upgradeKubernetesVersion"`
```

Comment:
I'm sure this was discussed offline, but I feel like having an extra config isn't what other suites do so far. i.e. provisioning, we made a provisioninginput folder for the config. 

I'm also having a hard time understanding why we would need the etcd object as part of this config, when this is used on existing clusters. 

Looking at the tests, it looks like the etcd config is being created / set on the cluster. But my understanding is, that should already be set on the cluster for these tests. Is there a reason we're requiring the user to enter it in the config vs. reading from the cluster?

Reply:
Talked offline about this, will update this struct to have only the needed values.